### ISSUE TITLE
Adding registration page

### ISSUE THREAD

**Issue by arindampradhan** [osu-cass/whats-fresh-api#56]
Currently, what's fresh does not have a registration page . http://whats-fresh-api.osuosl.org/login
It would be great if we have a registration page.
Steps:
Creating a user form for registration.
The template for the registration.
Adding the url for registration.

**Comment by Kennric** [comment]
The primary (currently only) use of the API application is to enter data for the mobile app, which we think is probably going to be done by a few people as part of an organization like Sea Grant, or a chamber of commerce, or a volunteer group. In this use case, a main admin at the organization would be able to add users, but it would not be open for just anyone to register and start adding data.

One of the long term goals for the application is to allow vendors to log in as well, and update their own information - a registration page would be useful there, but would need to be tied to the Vendor record, such that users who registered would get the correct access to the right records - so this is a bit bigger than just a registration page, and will require some careful thought and design.

I can also see a use case for the general public to log in, but only having access to very limited features - adding a review, perhaps, or reporting a vendor that no longer exists. Much of this is probably better done through the mobile app, which would also obviate the need for a registration page. This would require an expansion of the roles/permissions system to ensure correct access rights.

One last use case possibility is a fully community-operated database, where anyone can register and contribute to the knowledge base of products and vendors. This requires quite a bit of care and thought to make the system proof against spam (moderation, spam filtering, etc.) While I think this would be great eventually, I think that is more a version 3 or 4 than version 2.

In short, I think it is not a high priority to have a registration page, and any registration page would need to be part of a larger expansion of the roles/permissions system and other features.
